nix/pkgs.nix

```diff
@@ -0,0 +1,235 @@
+{ args ? { config = import ./config.nix; }
+, nixpkgs ? import <nixpkgs>
+}:
+let
+  pkgs = nixpkgs args;
+  overrideWith = override: default:
+   let
+     try = builtins.tryEval (builtins.findFile builtins.nixPath override);
+   in if try.success then
+     builtins.trace "using search host <${override}>" try.value
+   else
+     default;
+
+in
+let
+  # save the nixpkgs value in pkgs'
+  # so we can work with `pkgs` provided by modules.
+  pkgs' = pkgs;
+  #
+  nixpkgs' = import (import ./fetch-nixpkgs.nix) {};
+  # all packages from hackage as nix expressions
+  hackage = import (overrideWith "hackage"
+                    (nixpkgs'.fetchFromGitHub { owner  = "angerman";
+                                                repo   = "hackage.nix";
+                                                rev    = "d8e03ec0e3c99903d970406ae5bceac7d993035d";
+                                                sha256 = "0c7camspw7v5bg23hcas0r10c40fnwwwmz0adsjpsxgdjxayws3v";
+                                                name   = "hackage-exprs-source"; }))
+                   ;
+  # a different haskell infrastructure
+  haskell = import (overrideWith "haskell"
+                    (nixpkgs'.fetchFromGitHub { owner  = "angerman";
+                                                repo   = "haskell.nix";
+                                                rev    = "360fee0c28781d8554b6ec89e8d662f6002c37ab";
+                                                sha256 = "15m9f25n2y8hcf74cg2ggz6446kl13yvrgppiv1b27l9k3a3ss0k";
+                                                name   = "haskell-lib-source"; }))
+                   hackage;
+
+  # the set of all stackage snapshots
+  stackage = import (overrideWith "stackage"
+                     (nixpkgs'.fetchFromGitHub { owner  = "angerman";
+                                                 repo   = "stackage.nix";
+                                                 rev    = "67675ea78ae5c321ed0b8327040addecc743a96c";
+                                                 sha256 = "1ds2xfsnkm2byg8js6c9032nvfwmbx7lgcsndjgkhgq56bmw5wap";
+                                                 name   = "stackage-snapshot-source"; }))
+                   ;
+
+  # our packages
+  stack-pkgs = import ./.stack-pkgs.nix;
+
+  # Build the packageset with module support.
+  # We can essentially override anything in the modules
+  # section.
+  #
+  #  packages.cbors.patches = [ ./one.patch ];
+  #  packages.cbors.flags.optimize-gmp = false;
+  #
+  pkgSet = haskell.mkNewPkgSet {
+    inherit pkgs;
+    pkg-def = stackage.${stack-pkgs.resolver};
+    pkg-def-overlays = [
+      stack-pkgs.overlay
+      # We use some customized libiserv/remote-iserv/iserv-proxy
+      # instead of the ones provided by ghc. This is mostly due
+      # to being able to hack on them freely as needed.
+      #
+      # iserv is only relevant for template-haskell execution in
+      # a cross compiling setup.
+      {
+        ghci         = ./ghci.nix;
```

Comment:
@disassembler ideally `ghci` and `ghc-boot` would be on hackage, but they are not... because bootlibs are shipped with ghc supposedly and don't need to be on hackage.

The other three are due to the iserv-split patch, and again are not on hackage.

I really hope this will eventually be resolved and we'll get them all on hackage.
I wouldn't mind putting the binaries on S3 though. I just don't think I have any
access to publish blobs on S3.

Reply:
This is done.